pandas/_libs/parsers.pyx

```diff
@@ -683,15 +674,14 @@ cdef class TextReader:
         # header is now a list of lists, so field_count should use header[0]
 
         cdef:
-            Py_ssize_t i, start, field_count, passed_count, unnamed_count
+            Py_ssize_t i, start, field_count, passed_count, unnamed_count, level
             char *word
             object name, old_name
             uint64_t hr, data_line = 0
             char *errors = "strict"
             StringPath path = _string_path(self.c_encoding)
-
-        header = []
-        unnamed_cols = set()
+            list header = []
```

Comment:
Does this actually change anything? I'm surprised would have to declare this on both sides of the assignment

Reply:
If we have the line:

```cython
list header
```

instead of:

```cython
list header = []
```

```cython``` raises with 

> local variable 'header' referenced before assignment

---

Full error message:

<details>

```
Error compiling Cython file:
------------------------------------------------------------
...
                            hr -= 1
                            self.parser_start -= 1
                            this_header = [None] * lc

                data_line = hr + 1
                header.append(this_header)
               ^
------------------------------------------------------------

pandas/_libs/parsers.pyx:769:16: local variable 'header' referenced before assignment
Traceback (most recent call last):
  File "setup.py", line 791, in <module>
    setup_package()
  File "setup.py", line 761, in setup_package
    ext_modules=maybe_cythonize(extensions, compiler_directives=directives),
  File "setup.py", line 534, in maybe_cythonize
    return cythonize(extensions, *args, **kwargs)
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/Cython/Build/Dependencies.py", line 1102, in cythonize
    cythonize_one(*args)
  File "/home/bummy/miniconda3/envs/pandas-dev/lib/python3.7/site-packages/Cython/Build/Dependencies.py", line 1225, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: pandas/_libs/parsers.pyx
```

</details>